### DISCUSSION TITLE
Movie recording output suffers from dropped / skipped frames

### DISCUSSION THREAD

**Discussion by grrrrraham** [Motion-Project/motion#1774] [Q&A]
Hello, I have a camera streaming RTSP to Motion which is set to record when motion is detected. Quite often the recordings suffer from skipped or dropped frames throughout the recording. The camera has a time stamp with seconds in the top corner and I can see times where one or two and sometimes up to 5 to 10 seconds are skipped. 

I have tried all sorts of settings changes, from using passthrough video recording, transcoding at various quality and codec settings and nothing really seems to help.

My computer is high powered and not short on CPU resources. I have tried recording to a RAID of spinning disk as well as fast NVME flash drives. iotop and top show no resource contention.

Does anyone have any bright ideas as to why these recordings are suffering from dropped frames?

Also to note, if I stream the RTSP feed in VLC it streams just fine, no dropped frames so the frame drop is definitely happening between motion and the saved output files.

Thank you!

**Comment by tosiara** [comment]
is it tcp or udp?

**Reply by grrrrraham** [reply]
It appears to be TCP based on tcpdump

**Reply by grrrrraham** [reply]
TCP from the camera.. however I'm using media-mtx to distribute the stream since the camera crashes when too many streams happen at once. Looks like media-mtx uses UDP so consider it a UDP feed from media-mtx to motion.

**Comment by tosiara** [comment]
Also, a log with `-d 9` would be appreciated

**Reply by grrrrraham** [reply]
I have -d 9 enabled... waiting for some skipped frames to see if anything interesting comes

**Reply by grrrrraham** [reply]
Something of interest, it appears the incidents of dropped/skipped frames have been reduced quite a bit by setting "capture_rate" to 30 with a camera frame rate of 25. I noticed this in the syslog output (not d9 level mind you):

```
motion[804]: [6:nc6:Birdbox] [WRN] [NET] netcam_rtsp_connect: highres: capture_rate is less than camera FPS.
motion[804]: [6:nc6:Birdbox] [WRN] [NET] netcam_rtsp_connect: highres: capture_rate slightly larger than camera FPS is recommended.
```

**Reply by grrrrraham** [reply]
OK just had an event that lasted 35 seconds according to syslog output below. The resulting video was only 13 seconds with 2 seconds of black screen (another issue that comes up in some recordings) and then 11 seconds of motion recording. The time stamp of the camera starts at 13:58:16 and ends at 13:58:36 with a big skip from 13:58:21 to 13:58:30.

```
May 29 13:57:31 Mediacentre motion[275011]: [1:ml1:Birdbox] [NTC] [ALL] motion_detected: Motion detected - starting event 2.
May 29 13:57:31 Mediacentre motion[275011]: [1:ml1:Birdbox] [INF] [EVT] event_ffmpeg_newfile: Source FPS 25
May 29 13:57:31 Mediacentre motion[275011]: [1:ml1:Birdbox] [INF] [NET] util_check_passthrough: pass-through enabled.
May 29 13:57:31 Mediacentre motion[275011]: [1:ml1:Birdbox] [NTC] [ENC] ffmpeg_passthru_codec: pass-through mode enabled.  Changing to MP4 container.
May 29 13:57:31 Mediacentre motion[275011]: [1:ml1:Birdbox] [INF] [ENC] ffmpeg_passthru_codec: Pass-through stream opened
May 29 13:57:31 Mediacentre motion[275011]: [1:ml1:Birdbox] [NTC] [EVT] event_newfile: Writing movie to file: /media/Storage/Video/Recordings/100-02-20240529135730.mp4
May 29 13:58:06 Mediacentre motion[275011]: [1:ml1:Birdbox] [DBG] [EVT] event_closefile: Saved movie to file: /media/Storage/Video/Recordings/100-02-20240529135730.mp4
May 29 13:58:06 Mediacentre motion[275011]: [1:ml1:Birdbox] [NTC] [ALL] mlp_actions: End of event 2
```

**Reply by grrrrraham** [reply]
FWIW, I noticed a second camera (same model) of mine behaves quite well. Compared settings on the camera and noticed the problematic one was sending VBR video and the other well-behaved one was CBR.

I switched the other camera to CBR and it's almost perfect now. 

Not sure why this would cause issues with passthrough video file saving but it definitely seems to be correlated.

**Reply by tosiara** [reply]
Can you get us the FULL log with `-d 9` from the process start till end of an event where the issue has been reproduced for two cases: when CBR is set and when VBR is set?

**Reply by grrrrraham** [reply]
The logs I previously pasted are with -d 9. Are you expecting more verbosity? And for what its worth I don't see any difference in the logs for when events happen with dropped frames vs. not. 

If there's a way to get more verbosity that I'm missing please let me know. Thank you!

**Reply by tosiara** [reply]
Yes, I need exactly what I asked for:

FULL log with -d 9 from the process start till end of an event where the issue has been reproduced for two cases: when CBR is set and when VBR is set

**Comment by Menard01** [comment]
I can add that I suffer from the same weird thing, rarely but it happened, with an USB webcam (Logitech C525) I see myself dressing in front the door of my flat and in the following frames I am invisible. But it was in the default fps, 15, I elevated it to 25 then ... and survey...
